### PR TITLE
Fix Firefox video mimetype

### LINKED ISSUE #542: Add `mimeType` to File Uploader Options for camera recording
Different browsers support different mime types through the MediaRecorder object. If you add `mimeType` to the File Uploader options we could request a particular mime type from the browser for video recorded using MediaRecorder.

In Nov 2018 the mime type `video/webm` was hard-coded into the uploader widget then later changed to (I believe) accept the default mime type supplied by the browser.

https://github.com/uploadcare/uploadcare-widget/pull/516/files/d205b5bf3b071d7b9855f427cd03dbce70e7ab3f#diff-210dd7069c703b492ced317be0e77a42

Chrome used to deliver `video/webm` WebM files but now delivers `video/x-matroska` MKV files. It would be great if we could influence this with a `mimeType` setting.

### PR DESCRIPTION
**Pull request by jeetiss**
## Description

add hardcoded `mimeType` (`video/webm`) for Firefox
add `preferedMimeType` setting for camera tab

fix #542

## Checklist

- [ ] Tests (if applicable)
- [ ] Documentation (if applicable)
- [ ] Changelog stub (or use [conventional commit messages](https://www.conventionalcommits.org/))

### CODE REVIEW

File: src/settings.js [hunk 1]
```diff
@@ -72,6 +72,7 @@ defaults = {
   parallelDirectUploads: 10,
   passWindowOpen: false,
   // camera recording
+  preferedMimeType: null,
```
Comment (nd0ut) [inline src/settings.js:75]:
```suggestion
  preferredMimeType: null,
```

File: src/widget/tabs/camera-tab.js [hunk 2]
```diff
@@ -282,16 +282,28 @@ class CameraTab {
   }
 
   __startRecording() {
-    var __recorderOptions
     this.__setState('recording')
     this.__chunks = []
-    __recorderOptions = {}
+    var __recorderOptions = {}
+
+    var mimeType = this.settings.preferedMimeType
```
Comment (nd0ut) [inline src/widget/tabs/camera-tab.js:289]:
```suggestion
    var mimeType = this.settings.preferredMimeType
```

File: src/widget/tabs/camera-tab.js [hunk 3]
```diff
@@ -282,16 +282,28 @@ class CameraTab {
...
+    if (mimeType != null && this.MediaRecorder.isTypeSupported(mimeType)) {
+      __recorderOptions.mimeType = mimeType
+    }
+
+    var isFirefox = navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
+    if ( __recorderOptions.mimeType == null && isFirefox && this.MediaRecorder.isTypeSupported('video/webm')) {
```
Comment (nd0ut) [inline src/widget/tabs/camera-tab.js:295]: There is a space between `(` and `__recorderOptions`. Is it typo or linter?

File: src/settings.js (hunk 1 above)
Comment (homm) [inline src/settings.js:75]: This should be a list of ordered preferred options.
```suggestion
  videoPreferredMimeTypes: null,
```

**Comment by jeetiss** [comment]
hello @homm @nd0ut
I have adopted your suggestions, please check again

**Pull request merged**